qiskit_ibm_runtime/debugger/debugger.py

```diff
@@ -0,0 +1,170 @@
+
+from __future__ import annotations
+from typing import Optional, Sequence, List
+
+from qiskit_aer.noise import NoiseModel
+from qiskit_aer.primitives.estimator_v2 import EstimatorV2 as AerEstimator
+
+from qiskit.transpiler.passmanager import PassManager
+from qiskit.primitives.containers import EstimatorPubLike
+from qiskit.primitives.containers.estimator_pub import EstimatorPub
+from qiskit.providers import BackendV2 as Backend
+
+from qiskit_ibm_runtime.debugger.debugger_results import DebuggerResult
+from qiskit_ibm_runtime.transpiler.passes.cliffordization import ConvertISAToClifford
+from qiskit_ibm_runtime.utils import validate_estimator_pubs, validate_isa_circuits
+
+
+def _validate_pubs(
+    backend: Backend, pubs: List[EstimatorPub], validate_clifford: bool = True
+) -> None:
+    r"""Validates a list PUBs by running the :meth:`.~validate_estimator_pubs` and
+    :meth:`.~validate_isa_circuits` methods, and optionally, by checking if the PUBs
+    are Clifford.
+
+    Args:
+        backend: A backend.
+        pubs: A set of PUBs.
+        validate_clifford: Whether or not to validate that the PUB's circuit do not contain
+            non-Clifford gates.
+    """
+    validate_estimator_pubs(pubs)
+    validate_isa_circuits([pub.circuit for pub in pubs], backend.target)
+
+    if validate_clifford:
+        for pub in pubs:
+            cliff_circ = PassManager([ConvertISAToClifford()]).run(pub.circuit)
+            if pub.circuit != cliff_circ:
+                raise ValueError(
+                    "Given ``pubs`` contain a non-Clifford circuit. To fix, consider using the "
+                    "``ConvertISAToClifford`` pass to map your circuits to the nearest Clifford"
+                    " circuits, then try again."
+                )
+
+
+class Debugger:
+    r"""A class that users of the Estimator primitive can use to understand the expected
+    performance of their queries.
+
+    Args:
+        backend: A backend.
+        noise_model: A noise model for the operations of the given backend. If ``None``, it
+            defaults to the noise model generated by :meth:`NoiseModel.from_backend`.
+    """
+
+    def __init__(self, backend: Backend, noise_model: Optional[NoiseModel] = None) -> None:
+        self._backend = backend
+        self._noise_model = noise_model or NoiseModel.from_backend(
+            backend, thermal_relaxation=False
+        )
+
+    @property
+    def backend(self) -> Backend:
+        r"""
+        The backend used by this debugger.
+        """
+        return self._backend
+
+    @property
+    def noise_model(self) -> NoiseModel:
+        r"""
+        The noise model used by this debugger for the noisy simulations.
+        """
+        return self._noise_model
+
+    def simulate(
+        self,
+        pubs: Sequence[EstimatorPubLike],
+        with_noise: bool = True,
+        seed_simulator: Optional[int] = None,
+        default_precision: float = 0,
+    ) -> list[DebuggerResult]:
```

Comment:
What do you think about making the return type a class that can be sliced (similar to `PrimitiveResult`), instead of a list? A class would give us a lot more flexibility if we ever want to change the output type, or include additional metadata.

Reply:
I am not opposed to that. But we need to find one more name

Reply:
`DebuggerResult` which has a list of `DebuggerPubResult` 😄

Reply:
Okay. I like this, because all we need to change is "Debugger*" (if we want to), otherwise it's all pretty consistent with Estimator